cropduster/resizing.py

```diff
@@ -371,79 +363,23 @@ def best_fit(self, w=None, h=None, min_w=None, min_h=None, max_w=None, max_h=Non
 
         return Crop(Box(x1, y1, x2, y2), self.image)
 
-    def add_xmp_to_crop(self, cropped_image, size, original_image=None):
-        from cropduster.utils import get_image_extension
+    def add_xmp_to_crop(self, cropped_image_path, size, original_image=None):
         try:
-            from cropduster.standalone.metadata import libxmp, file_format_supported
+            from cropduster.standalone.metadata import libxmp, get_file_xmp, put_file_xmp
         except ImproperlyConfigured:
             libxmp = None
 
-        if not libxmp:
-            return
-
-        from cropduster.models import Image, Thumb
-
-        def get_image_path(img):
-            if isinstance(img, ImageFile):
-                path = img.filename
-            elif isinstance(img, BUILTIN_FILE_TYPE):
-                path = img.name
-            elif isinstance(img, FieldFile):
-                path = img.path
-            elif isinstance(img, Thumb):
-                image = img.image
-                if not img:
-                    return None
-                path = image.get_path(img.name)
-            elif isinstance(img, Image):
-                path = img.get_path('original')
-            elif isinstance(img, six.string_types):
-                path = img
-            else:
-                path = None
-            return path
-
-        image_path = get_image_path(cropped_image)
-
-        if not image_path:
+        if not libxmp or not cropped_image_path:
             return
 
-        temp_file = tempfile.NamedTemporaryFile(suffix=get_image_extension(self.image), delete=False)
-        with default_storage.open(image_path, mode='rb') as f:
-            temp_file.write(f.read())
-        temp_file.seek(0)
-        xmp_file = libxmp.XMPFiles(file_path=temp_file.name, open_forupdate=True)
-
-        original_metadata = None
         if original_image and CROPDUSTER_RETAIN_METADATA:
-            original_image_path = get_image_path(original_image)
-            if original_image_path:
-                original_temp_file = tempfile.NamedTemporaryFile(suffix=get_image_extension(original_image), delete=False)
-                with default_storage.open(original_image_path, mode='rb') as f:
-                    original_temp_file.write(f.read())
-                try:
-                    original_xmp_file = libxmp.XMPFiles(file_path=original_temp_file.name)
-                    original_metadata = original_xmp_file.get_xmp()
-                except:
-                    pass
+            original_metadata = get_file_xmp(original_image.get_path('original'))
```

Comment:
I think this needs to be `original_metadata = get_file_xmp(original_image.filename)`, from a local test